tests/testlib/s2n_random_test_utils.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <sys/param.h>
+
+#include "utils/s2n_mem.h"
+#include "utils/s2n_random.h"
+#include "utils/s2n_safety.h"
+#include "crypto/s2n_drbg.h"
+#include "tests/testlib/s2n_testlib.h"
+
+#if S2N_LIBCRYPTO_SUPPORTS_CUSTOM_RAND
+
+static uint8_t seed_buffer[S2N_DRBG_MAX_SEED_SIZE] = { 0 };
+
+S2N_RESULT s2n_fixed_entropy_generator(struct s2n_blob *blob)
+{
+    ENSURE_LTE(blob->size, S2N_DRBG_MAX_SEED_SIZE);
+    blob->data = seed_buffer;
+    return S2N_RESULT_OK;
+}
+
+static struct s2n_drbg fixed_drbg = { .entropy_generator = &s2n_fixed_entropy_generator };
+
+int s2n_unsafe_drbg_reseed(uint8_t *seed, uint8_t seed_size)
```

Comment:
Should we put in `S2N_ERROR_IF(!s2n_in_unit_test(), S2N_ERR_NOT_IN_UNIT_TEST);` to ensure this doesn't get used improperly?

Reply:
Nevermind, you beat me to it :D